### PR TITLE
Update routing table via Chitchat

### PR DESCRIPTION
**Pull request by guilload**
### Description
Per title.

### How was this PR tested?
- Added unit tests
- Test manually

### CODE REVIEW

File: quickwit/quickwit-ingest/src/ingest_v2/fetch.rs [hunk 1]
```diff
@@ -645,7 +645,7 @@ pub(super) mod tests {
             .unwrap();
         drop(state_guard);
 
-        let fetch_message = timeout(Duration::from_millis(50), fetch_stream.next())
+        let fetch_message = timeout(Duration::from_millis(100), fetch_stream.next())
```
Comment (guilload) [inline quickwit/quickwit-ingest/src/ingest_v2/fetch.rs:648]: Got a flaky test with 50 ms. Increased to 100.

File: quickwit/quickwit-ingest/src/ingest_v2/mod.rs [hunk 2]
```diff
@@ -26,7 +26,7 @@ mod mrecordlog_utils;
 mod rate_meter;
 mod replication;
 mod router;
-mod shard_table;
+mod routing_table;
```
Comment (guilload) [inline quickwit/quickwit-ingest/src/ingest_v2/mod.rs:29]: Renamed `shard_table::ShardTable` to `routing_table::RoutingTable`.

File: quickwit/quickwit-indexing/Cargo.toml [hunk 3]
```diff
@@ -20,7 +20,6 @@ async-trait = { workspace = true }
 backoff = { workspace = true, optional = true }
 bytes = { workspace = true }
 bytesize = { workspace = true }
-chitchat = { workspace = true }
```
Comment (fulmicoton) [inline quickwit/quickwit-indexing/Cargo.toml:23]: awesome

File: quickwit/quickwit-ingest/src/ingest_v2/router.rs [hunk 4]
```diff
@@ -396,22 +394,32 @@ impl EventSubscriber<LocalShardsUpdate> for Weak<RwLock<RouterState>> {
...
-            .shard_table
+
+        let mut open_shard_ids: Vec<_> = Vec::new();
+        let mut closed_shard_ids: Vec<ShardId> = Vec::new();
+
+        for shard_info in local_shards_update.shard_infos {
+            match shard_info.shard_state {
+                ShardState::Open => open_shard_ids.push(shard_info.shard_id),
+                ShardState::Closed => closed_shard_ids.push(shard_info.shard_id),
+                _ => {}
```
Comment (fulmicoton) [inline quickwit/quickwit-ingest/src/ingest_v2/router.rs:408]:
```suggestion
                _ => {}
```

nitpick, it is safer to put the other state than using `_` here.

Also, do we want to do something on Unavailable?

File: quickwit/quickwit-ingest/src/ingest_v2/routing_table.rs [hunk 5]
```diff
@@ -0,0 +1,835 @@
...
+    }
+
+    /// Closes the shards identified by their shard IDs.
+    fn close_shards(&mut self, index_uid: &IndexUid, shard_ids: &[ShardId]) {
+        // If the shard table was just recently updated with shards for a new index UID, then we can
+        // safely discard this request.
+        if self.index_uid != *index_uid {
+            return;
+        }
+        for shards in [&mut self.local_shards, &mut self.remote_shards] {
```
Comment (fulmicoton) [inline quickwit/quickwit-ingest/src/ingest_v2/routing_table.rs:230]: should we just entirely remove the entry if we are in the remote shards?
If not, when will be removed?

I forgot the other PR. Do we cleanup this metadata for local shard on EOF truncate?

**Review by fulmicoton — APPROVED** [review]
Approving, but the rename did not work too well. I could not see the diff between routing table and shard table.

File: quickwit/quickwit-ingest/src/ingest_v2/routing_table.rs (hunk 5 above)
Comment (guilload) [reply quickwit/quickwit-ingest/src/ingest_v2/routing_table.rs:230]: I'm currently adding the removal of deleted shards from the routing table to this PR.

File: quickwit/quickwit-ingest/src/ingest_v2/routing_table.rs [hunk 6]
```diff
@@ -0,0 +1,835 @@
...
+                if shard.shard_state.is_open() && ingester_pool.contains_key(&shard.leader_id) {
+                    return Some(shard);
+                }
+            }
+        }
+        None
+    }
+
+    /// Inserts the open shards the routing table is not aware of.
+    fn insert_open_shards(
```
Comment (guilload) [inline quickwit/quickwit-ingest/src/ingest_v2/routing_table.rs:160]: @fulmicoton, this is the function I added to the routing table in this PR. This is where you should focus your review.

**Pull request merged**